### DISCUSSION TITLE
how to catch indirect relation existence

### DISCUSSION THREAD

**Discussion by Qa5imm** [pocketbase/pocketbase#3647] [Q&A]
hello,
Im using pocket base for building an app, I have three tables `users` , `skills`,`courses`. When I query data from pocket base for user such that

pb.collection('users').getOne(uid, {
expand: 'skills(user), courses(user)'
})

it works fine when user has both skills and courses record in respective collections. however, if a user don't have record in one of the collections or both, it throws error saying relation don't exist.

is there any way to cater to like in sql it returns empty in case a relation record don't exist

thank you!!

**Comment by ganigeorgiev** [comment]
I'm not able to reproduce it. In case of a missing expand relation you should get only a print in the console if `--debug` is enabled but it shouldn't throw an error.

Make sure that your request satisfies the View API rule of the "users" collection.

If you are still not able to resolve the issue, please paste your collections configuration as JSON (Admin UI > Settings > Export collections) to try to reproduce it.

**Reply by Qa5imm** [reply]
can you please let me know how to set the `--debug` flag, I couldn't find it in the docs. Moreover this is the error that I got 
 `error ClientResponseError 404: The requested resource wasn't found`

**Reply by ganigeorgiev** [reply]
To enable the debug mode you can append it to the `serve` command, aka. `./pocketbase serve --debug`.

But in your case the error is most likely caused by unsatisfied View API rule.

_If after checking your View API rule you are still not able to identify the culprit, please export your collections configuration as json (Admin UI > Settings > Export collections)._

**Reply by Qa5imm** [reply]
I am not in a position to share the collections (is there any way I can share only  the schema without any data in it? that I can do), apologies for that. My View rule for users collection is empty what should I change it to make it work

**Reply by ganigeorgiev** [reply]
_Admin UI > Settings > Export collections_ will export only the collections configuration without records data.

If your users View API rule is empty (aka. allowed for everyone) then make sure that the `uid` you are searching for actually exists.

**Reply by Qa5imm** [reply]
apologies but can you let me know what should I write in the user field in api rule to make it work or I have to check it manually, I am new to pocket base so not sure. I have looked in the docs but couldn't find anything about it. Plus can you link to some helpful resource which I can look up to when I am stuck, the docs don't contain much examples or I am missing some section.

**Reply by Qa5imm** [reply]
resolved, it was an issue on frontend.

thank you @ganigeorgiev
